### PR TITLE
Fix restore_backup()

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix restore_backup() when no secret manager data is stored inside

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Added test and cli wallet

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/tests/wallet/backup_restore.rs [hunk 1]
```diff
@@ -527,3 +527,44 @@ async fn backup_and_restore_bech32_hrp_mismatch() -> Result<()> {
     );
     tear_down(storage_path)
 }
+
+#[tokio::test]
+// Restore a Stronghold snapshot without secret manager data
```
Comment (kwek20) [inline sdk/tests/wallet/backup_restore.rs:531]: nit but we always but the comment above ;)

Comment (Thoralf-M) [reply sdk/tests/wallet/backup_restore.rs:531]: Not always as you can see in the commit :trollface: But changed

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/nodejs/CHANGELOG.md [hunk 2]
```diff
@@ -19,11 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
-## 1.0.6 - 2023-MM-DD
+## 1.0.6 - 2023-08-25
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md:22]: Also wasm?

Comment (Thoralf-M) [reply bindings/nodejs/CHANGELOG.md:22]: No, wasm has no stronghold

Comment (thibault-martinez) [reply bindings/nodejs/CHANGELOG.md:22]: Right

File: sdk/src/wallet/core/operations/stronghold_backup/mod.rs [hunk 3]
```diff
@@ -140,6 +140,10 @@ impl Wallet {
                 stronghold.set_password(stronghold_password).await?;
             }
             *secret_manager = restored_secret_manager;
+        } else {
+            // If no secret manager data was in the backup, just copy the Stronghold file so the seed is available in
+            // the new location
```
Comment (thibault-martinez) [inline sdk/src/wallet/core/operations/stronghold_backup/mod.rs]:
```suggestion
            // the new location.
```

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**